File: Provenance/Controller/iCade/PViCadeControllerViewController.m
//
//  PViCadeControllerViewController.m
//  Provenance
//
//  Created by James Addyman on 17/04/2015.
//  Copyright (c) 2015 James Addyman. All rights reserved.
//

#import "PViCadeControllerViewController.h"
#import "kICadeControllerSetting.h"
#import "PVSettingsModel.h"

@implementation PViCadeControllerViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
#if TARGET_OS_TV
    [self.splitViewController setTitle:@"Supported iCade Controllers"];
    [self.tableView setBackgroundColor:[UIColor clearColor]];
    [self.tableView setBackgroundView:nil];
#else
    self.title = @"Supported iCade Controllers";
#endif
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return kICadeControllerSetting_Count;
}

- (nullable NSString *)tableView:(UITableView *)tableView titleForFooterInSection:(NSInteger)section
{
    return @"Your controller must be paired with your device in order to work";
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"Cell"];
    if (!cell)
    {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"Cell"];
    }
    if ([indexPath row] == [[PVSettingsModel sharedInstance] iCadeControllerSetting]) {
        [cell setAccessoryType:UITableViewCellAccessoryCheckmark];
    } else {
        [cell setAccessoryType:UITableViewCellAccessoryNone];
    }
    [[cell textLabel] setText:kIcadeControllerSettingToString((kICadeControllerSetting)[indexPath row])];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [[self.tableView cellForRowAtIndexPath:indexPath] setAccessoryType:UITableViewCellAccessoryCheckmark];
    [self.tableView deselectRowAtIndexPath:[self.tableView indexPathForSelectedRow] animated:YES];
    PVSettingsModel *settings = [PVSettingsModel sharedInstance];
    [settings setICadeControllerSetting:[indexPath row]];
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: Provenance/Controller/iCade/PViCadeControllerViewController.swift
//  Converted to Swift 4 by Swiftify v4.1.6640 - https://objectivec2swift.com/
//
//  PViCadeControllerViewController.swift
//  Provenance
//
//  Created by James Addyman on 17/04/2015.
//  Copyright (c) 2015 James Addyman. All rights reserved.
//

import UIKit

class PViCadeControllerViewController: UITableViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
#if TARGET_OS_TV
        tableView.backgroundColor = UIColor.clear
        tableView.backgroundView = nil
#endif
    }

    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return iCadeControllerSetting.settingCount.rawValue
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "iCadeCell") ?? UITableViewCell(style: .default, reuseIdentifier: "iCadeCell")

        if indexPath.row == PVSettingsModel.shared.myiCadeControllerSetting.rawValue {
            cell.accessoryType = .checkmark
        } else {
            cell.accessoryType = .none
        }
        cell.textLabel?.text = iCadeControllerSettingToString((iCadeControllerSetting(rawValue: indexPath.row))!)

#if os(iOS)
        cell.textLabel?.textColor = Theme.currentTheme.settingsCellText
#endif

        return cell
    }

    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return "Supported iCade Controllers"
    }

    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
        return "Controllers must be paired with device."
    }

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        self.tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
        if let aRow = self.tableView.indexPathForSelectedRow {
            self.tableView.deselectRow(at: aRow, animated: true)
        }
        PVSettingsModel.shared.myiCadeControllerSetting = iCadeControllerSetting(rawValue: indexPath.row)!
        PVControllerManager.shared.resetICadeController()
        navigationController?.popViewController(animated: true)
    }
}
